core/src/main/scala/cats/FlatMap.scala

```diff
@@ -90,4 +91,16 @@ import simulacrum.typeclass
    */
   def ifM[B](fa: F[Boolean])(ifTrue: => F[B], ifFalse: => F[B]): F[B] =
     flatMap(fa)(if (_) ifTrue else ifFalse)
+
+  /**
+   * Keeps calling `f` until a `[[cats.data.Xor.Right Right]][B]` is returned.
+   *
+   * Based on Phil Freeman's
+   * [[http://functorial.com/stack-safety-for-free/index.pdf Stack Safety for Free]].
+   *
+   * Implementations of this method should ideally use constant stack space. If
+   * it is constant stack space, an instance of `RecursiveTailRecM[F]` should
+   * be made available.
+   */
+  def tailRecM[A, B](a: A)(f: A => F[A Xor B]): F[B]
```

Comment:
Can we add a comment saying theres a `defaultTailRecM` that could be used as a stub/template?